llama_stack/providers/remote/inference/groq/groq.py

```diff
@@ -0,0 +1,133 @@
+from llama_models.llama3.api.datatypes import (
+    InterleavedTextMedia,
+    Message,
+    ToolChoice,
+    ToolDefinition,
+    ToolPromptFormat,
+)
+from llama_models.datatypes import SamplingParams
+from llama_stack.apis.inference import (
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseEvent,
+    ChatCompletionResponseStreamChunk,
+    ChatCompletionResponseEventType,
+    CompletionResponse,
+    CompletionMessage,
+    CompletionResponseStreamChunk,
+    EmbeddingsResponse,
+    Inference,
+    LogProbConfig,
+    ResponseFormat,
+)
+from llama_stack.providers.utils.inference.model_registry import (
+    build_model_alias,
+    ModelRegistryHelper,
+)
+from llama_stack.providers.remote.inference.groq.config import GroqConfig
+from llama_models.sku_list import CoreModelId
+from llama_models.llama3.api.datatypes import StopReason
+from groq import Groq
+
+_MODEL_ALIASES = [
+    build_model_alias(
+        "llama3-8b-8192",
+        CoreModelId.llama3_2_3b_instruct.value,
+    ),
+]
+
+class GroqInferenceAdapter(Inference, ModelRegistryHelper):
+    client: Groq
+    def __init__(self, config: GroqConfig) -> None:
+        ModelRegistryHelper.__init__(self, model_aliases=_MODEL_ALIASES)
+        self.client = Groq(api_key=config.api_key)
+
+    def completion(
+        self,
+        model_id: str,
+        content: InterleavedTextMedia,
+        sampling_params: Optional[SamplingParams] = SamplingParams(),
+        response_format: Optional[ResponseFormat] = None,
+        stream: Optional[bool] = False,
+        logprobs: Optional[LogProbConfig] = None,
+    ) -> Union[CompletionResponse, AsyncIterator[CompletionResponseStreamChunk]]:
+        raise NotImplementedError()
+
+    async def embeddings(
+        self,
+        model_id: str,
+        contents: List[InterleavedTextMedia],
+    ) -> EmbeddingsResponse:
+        raise NotImplementedError()
+
+    async def chat_completion(
+        self,
+        model_id: str,
+        messages: List[Message],
+        sampling_params: Optional[SamplingParams] = SamplingParams(),
+        response_format: Optional[ResponseFormat] = None,
+        tools: Optional[List[ToolDefinition]] = None,
+        tool_choice: Optional[ToolChoice] = ToolChoice.auto,
+        tool_prompt_format: Optional[
+            ToolPromptFormat
+        ] = None,  # API default is ToolPromptFormat.json, we default to None to detect user input
+        stream: Optional[bool] = False,
+        logprobs: Optional[LogProbConfig] = None,
+    ) -> Union[
+        ChatCompletionResponse, AsyncGenerator
+    ]:
+        if stream:
+            response = self.client.chat.completions.create(
+                model=self.get_provider_model_id(model_id),
+                messages=messages,
+                stream=True,
+            )
+            async def stream_response():
+                for chunk in response:
+                    if chunk.choices[0].finish_reason == 'stop':
```

Comment:
Groq only supports n=1 at this stage:

<img width="655" alt="image" src="https://github.com/user-attachments/assets/a74138b3-afae-4e81-96bc-5cbca6e480f4">